### DISCUSSION TITLE
Use `ESLint` directly instead of `xo`

### DISCUSSION THREAD

**Discussion by JounQin** [mdx-js/mdx#2065] [General] [answered]
It works differently with ESLint.

https://github.com/xojs/xo/issues/671

It is slow

https://github.com/xojs/xo/issues/515
https://github.com/xojs/xo/issues/558

And no way to debug

https://github.com/xojs/xo/issues/188

No `root` option, so global ESLint config like `~/.eslintrc` will be used unexpectedly.

And all its functionality can be replaced by `ESLint` directly.

@wooorm

**Comment by wooorm** [comment, marked as answer]
I don’t quite understand.

- Different: It being different, namely that it has useful features, is why I use xo. It looks like in 671, there are rules that xo turns on, which ESLint doesn’t default to, which don’t work with `eslint-mdx`?
- Slow: It can be slow, but that’s because it includes a lot of useful rules
- Debugging: My guess is that it has the same debugging as ESLint: `DEBUG="*" xo...`
- Root: The `root` option you mention seems to be a problem with VS Code? If possible, I recommend not using a global `~/.eslintrc`. Otherwise, you can typically tell VS Code plugins that they need to be configured in a project. I believe xo has such an option

> And all its functionality can be replaced by ESLint directly.

I find the functionality that xo provides useful and I don’t want to manually handle ESLint

**Reply by JounQin** [reply]
1. `no-unused-expressions` is already enabled in `.eslintrc`
4. No issue with `VS Code`, I have a global `~/.eslintrc` because I want to enable `eslint` for all my files even when I'm not developing a serious project, so without a project `.eslintrc`. Or a non-js project.

> I believe xo has such an option

<img width="721" alt="image" src="https://user-images.githubusercontent.com/8336744/174650959-155555ed-1961-4f42-a66c-071bc621460f.png">

It seems not.

**Reply by wooorm** [reply]
1\. I don’t understand you.
3\. I meant the vscode xo integration. Where are you using xo without a xo config?

**Reply by JounQin** [reply]
1. `no-unused-expressions` is enabled in `.eslintrc` of https://github.com/xojs/xo/issues/671, but it results different

**Reply by JounQin** [reply]
```sh
npx xo --print-config test.mdx > .eslintrc.json
```

Seems the difference is

```json
{
  "no-unused-expressions": [
  2,
  {
    "enforceForJSX": true,
    "allowShortCircuit": false,
    "allowTernary": false,
    "allowTaggedTemplates": false
    }
  ]
}
```

**Reply by wooorm** [reply]
Perhaps `eslint-mdx` can be made to work with that ESLint configuration?

**Reply by JounQin** [reply]
Maybe, I need to read `no-unused-expressions` source code tomorrow.

Going to sleep now, seems I'm not very sober at this moment.

**Reply by wooorm** [reply]
good night! ;)
